applications/StructuralMechanicsApplication/custom_constitutive/linear_j2_plasticity_3d.cpp

```diff
@@ -0,0 +1,452 @@
+{
+    if(rThisVariable == INELASTIC_FLAG){
+        rValue = mInelasticFlag;
+    }
+
+    return rValue;
+}
+
+void LinearJ2Plasticity3D::InitializeMaterial(const Properties& material_prop,
+                                              const GeometryType& rElementGeometry,
+                                              const Vector& rShapeFunctionsValues)
+{
+    mPlasticStrainOld = ZeroVector(this->GetStrainSize());
+    mAccumulatedPlasticStrainOld = 0.0;
+}
+
+void LinearJ2Plasticity3D::FinalizeSolutionStep(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    mPlasticStrainOld = mPlasticStrain;
+    mAccumulatedPlasticStrainOld = mAccumulatedPlasticStrain;
+}
+
+void LinearJ2Plasticity3D::CalculateMaterialResponsePK1(Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+void LinearJ2Plasticity3D::CalculateMaterialResponsePK2(Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+void LinearJ2Plasticity3D::CalculateMaterialResponseKirchhoff(Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+void LinearJ2Plasticity3D::CalculateMaterialResponseCauchy(Parameters& rValues)
+{
+    Flags &Options = rValues.GetOptions();
+
+    const Properties& rMaterialProperties = rValues.GetMaterialProperties();
+    Vector& StrainVector = rValues.GetStrainVector();
+    Vector& StressVector = rValues.GetStressVector();
+
+
+    if (Options.IsNot(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN)) {
+    }
+
+    if (Options.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR)) {
+        Matrix& ConstitutiveMatrix = rValues.GetConstitutiveMatrix();
+        CalculateElasticMatrix(ConstitutiveMatrix, rMaterialProperties);
+    }
+
+    if (Options.Is(ConstitutiveLaw::COMPUTE_STRESS)) {
+        if (rValues.GetProcessInfo().Has(INITIAL_STRAIN)) {
+            noalias(StrainVector) += rValues.GetProcessInfo()[INITIAL_STRAIN];
+        }
+        Matrix ElasticTensor;
+        Matrix& TangentTensor = rValues.GetConstitutiveMatrix();
+        const double hardening_modulus = rMaterialProperties[ISOTROPIC_HARDENING_MODULUS];
+        const double delta_k = rMaterialProperties[INFINITY_HARDENING_MODULUS];
+        const double hardening_exponent = rMaterialProperties[HARDENING_EXPONENT];
+        double trial_yield_function;
+        const double E = rMaterialProperties[YOUNG_MODULUS];
+        const double poisson_ratio = rMaterialProperties[POISSON_RATIO];
+        const double mu = E / (2. + 2. * poisson_ratio);
+        const double volumetric_modulus = E / (3. * (1. - 2. * poisson_ratio));
+
+        mPlasticStrain = mPlasticStrainOld;
+        mAccumulatedPlasticStrain = mAccumulatedPlasticStrainOld;
+
+        ElasticTensor.resize(6, 6);
+        CalculateElasticMatrix(ElasticTensor, rMaterialProperties);
+        Vector sigma_trial;
+        sigma_trial.resize(6);
```

Comment:
better do
Vector sigma_trial(6)

the point is that if you omit the "false" in the resize, it copies the old values